### ISSUE TITLE
5$: Improve error message for '#foo = bar;'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#371]
`#foo = bar;` produces two errors, but it should produce only one:
```js
#myvar = 42;
```
```
hello.js:1:1: error: cannot reference private variables without object; use 'this.' [E155]
hello.js:1:1: error: invalid expression left of assignment [E020]
```

We should omit E020, and only report E155.

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by singalhimanshu** [comment]
Fixed in #372
